### DISCUSSION TITLE
Chance for proxy/original workflow?

### DISCUSSION THREAD

**Discussion by tangierc** [latenitefilms/BRAWToolbox#14] [Feature Requests]
Hello Chris. One of the things I was particularly hopeful for is just bringing in Braw video that I can use in a proxy/original workflow and allow FCP or Sync-N-Link to do my syncing using XML. A large part of this equation is being able to relink clips as needed, but BRAW Toolbox imports are don't allow me access to FCP's native reveal in browser, reveal, in finder, and relinking capabilities. 

My proxies were already made using Premiere Pro and migrating the project to FCP, I wanted to link those proxies to their original counterpart using BRAW Toolbox and in the end send to Resolve for color. Would these capabilities be possible in a future update? I know this is a beta. As it stands I am unable to use BRAW Toolbox imports both due to playback and scrubbing performance issues (on an Intel Mac) and no access to FCP's native clip management features.

Thank you for all of your hard work.

**Comment by latenitefilms** [comment]
Apologies for the delayed reply - I totally forgot to reply.

Currently there's no way to convert "regular clips" to BRAW Toolbox clips - but this is definitely something I want to explore in the coming weeks.

Have you actually got footage inside Premiere, or did you just use it to make proxies?

One of the main advantages of BRAW Toolbox is that you don't need to make proxies - you can just edit using a low Decode Quality, then when it comes time to export, you can change to the Full Decode Quality. Even on lower end machines, you should still be able to get real-time playback on a lower Decode Quality setting.

If you have dual-system audio, you should eventually be able to use tools like Sync-N-Link with the synchronised clips and your location audio. It looks like there's currently maybe a bug in Sync-N-Link preventing this, but I hope to resolve this soon.

In terms of re-linking, that's a bit more complicated. I'll be adding a Toolbox feature called "Relink Unliked BRAW Clips within an EVENT" at some point soon, so you can relink BRAW clips within an event. However, an easier workaround, is to just keep the files on an external drive with the same name.

Let me know if you're still having issues with playback and scrubbing performance with Beta 2. It's pretty smooth and reliable on my MacBook Pro (16-inch, 2021, Apple M1 Max, 64 GB RAM, macOS 12.5.1, FCPX 10.6.5). I'll do some more testing with my Intel 16-inch MacBook Pro.

**Reply by latenitefilms** [reply]
FYI: Sync-N-Link v1.1.4 fixes the Synchronised Clips bug.

**Comment by tangierc** [comment]
Thanks for the response Chris.  I already had the proxies given to me. There was an edit done in Premiere Pro using a combination of ProRes LT proxies made in-camera by a Red V-Rapter and Scarlet, and some BMPCC6K BRAW clips. I am migrating the project out of Premiere Pro into FCP using SendToX; right now using the proxies for the BRAWs being the only option.

I tested it on a 2013 Mac Pro using a 6K BRAW clip and BRAW Toolbox Decode quality set to Eighth (toggle enabled). FCP 10.6.5 playback settings are set to Better Performance.  
After processing the clip, playback is very choppy and not useable for playback or editing. It takes a moment for the frames to draw in icon view in the browser and playing in the browser will also give me flickering blue screens. I know this computer is capable being that the same BRAW clip in Resolve 18 plays and scrubs quickly without issue.  The BRAW clip is on an OWC Gemini in RAID 1 over thunderbolt 2.

If I can get smooth playback from the BRAW in FCP, then I’ll be able to leave FCP in Proxy preferred playback mode to accommodate both the BRAW clips and the proxies for the RED clips.

I attempted to sync my testing Red, BRAW clip from BAW Toolbox, and a WAV file using Sync-N-Link. I did get an error. I presume this is what you were talking about.

I have a 14-inch M1 Pro MacBook Pro, but I won’t be editing this project off of that, not mostly at least. I haven’t tested BRAW Toolbox on there.

Hope this helps.

Thanks again.

Sincerely,


Tangier

**Comment by tangierc** [comment]
Hello Chris. Just thought I’d mention that before my last post I did create a new library first and had scrubbing and playback issues.
I decided to make another new library and restart FCP and the scrubbing of the BRAW clip at Eighth decode was great just for the clip in the browser and the timeline.
What’s strange is that now scrubbing is just about fluid with these settings, but just playing it back I get flashing blew frames once I set the playhead somewhere before it starts to play and then the playback seems like it’s struggling.

Tangier Clarke

**Reply by latenitefilms** [reply]
What GPUs do you have in the MacPro? Have you tried trashing FCPX's preferences?

Can you try BRAW Toolbox on your MacBook Pro M1 just to see if it plays fine there?

Have you tried some other BRAW clips? What camera are these BRAW files from? Are you able to share a clip?

**Reply by latenitefilms** [reply]
Also, can you share a screen recording of what you're seeing?

Blue frames is... odd. It sounds like a GPU issue, but it's weird then that the same files play back fine in DaVinci Resolve.

**Comment by tangierc** [comment]
Chris, I’ll send a screen recording. How can send you a private link outside of Github being that this is footage currently in use for a project?

FCP Preferences have bee trashed a few times since I’ve been working on another FCP UI responsiveness issue on larger projects here with other people on FCP.co <http://fcp.co/>. You may find some interesting tidbits here:
https://fcp.co/forum/4-final-cut-pro-x-fcpx/37612-choppy-timeline-editing-getting-back-to-fluidity <https://fcp.co/forum/4-final-cut-pro-x-fcpx/37612-choppy-timeline-editing-getting-back-to-fluidity>

GPUs are Dual AMD FirePro D500

Sometimes playback and performance is hit and miss. A clip that plays fine after the first import to the library will play worse later. I can’t explain this.

In the screen recording I have, the flickering is blue, but sometime it’ll be green.

I tried other BRAW clips. It’s more of the same. 

BRAW clips have LTC in them. That shouldn’t be an issue, but thought I’d mention it anyway.

I have no A/V output also. Just two desktop monitors. Thought I’d mention that just in case there’s a resource issue.

I moved the footage to the internal drive so that it’s on the SSD in order to test and see if it’s an issue with the connection between the OWC RAID and computer. There results were the same even on the internal drive.

Tested on an M1 Pro MacBook Pro on its internal SSD. No issues at all during playback or scrubbing. 

Thanks,

Tangier

**Reply by latenitefilms** [reply]
You can send me files here:

https://latenitefilms.digitalpigeon.com

**Reply by latenitefilms** [reply]
Could you also please send the log files?

https://github.com/latenitefilms/BRAWToolbox#ive-run-into-a-bug-where-can-i-find-the-log-files

**Comment by tangierc** [comment]
Will do. All files are on their way uploading to digitalpigeon now.


Tangier Clarke
Motion Picture Editor / IT Admin
Teacher / Da Vinci Communications
http://tangierclarke.com
***@***.***

**Reply by latenitefilms** [reply]
Legend, thank you! It seems that for some reason, the BRAW SDK is regularly failing to generate BRAW frames on your machine - I'm not really sure why yet. I assume it has something to do with the Dual GPUs on the MacPro - as I haven't had any other reports of this, although most of our users are on machines with a single GPU (MacBook Pro's, Mac mini's, Mac Studio's, etc).

I've reached out to Blackmagic to see if they have any ideas.

I know a lot of software had to do updates to make things work nicely with Dual GPUs on the MacPro - so maybe there's something specific we need to do to make it all work nicely.

I've created a new issue here: https://github.com/latenitefilms/BRAWToolbox/issues/25

**Comment by tangierc** [comment]
Thanks Chris. Yeah it is strange the performance (on the 2013 Mac Pro) is so poor with the BRAW clips in FCP, but not in Resolve on the same machine. Though it has two GPUs, my understanding is that one handles graphics calculations while the other handles the signal to the displays. That’s what I’ve gathered from some Apple online support communities. That’s not to say that one of the GPUs isn’t enough to handle the task, but I can’t say that for sure.

Curious your findings if you find out.

Thanks,

Tangier

**Reply by latenitefilms** [reply]
I definitely don't think it's a GPU power issue - a single AMD FirePro D500 can easily handle BRAW processing at a low decode quality, as you can see in Resolve. I'm GUESSING the problem is probably the BRAW SDK rendering on one GPU, and FCPX rendering on another GPU, and for whatever reason something is going funky moving data between the two GPUs and RAM.

**Comment by latenitefilms** [comment]
One question I've never thought to ask... what's the playback performance like in the Workflow Extension?

**Comment by tangierc** [comment]
I hadn’t considered comparing the two either. Scrubbing is very fluid. Playback is smooth as well. The one caveat is that if I jump around in the playback there is a slight momentary difference in frame rate as if it’s ramping up to play properly. It’s very slight. Though both are good inside that workflow extension as far as I can tell trying it with with multiple clips shot on different days including the ones I sent you.

Tangier

**Reply by latenitefilms** [reply]
Ok, so it seems there's no issue with BRAW playback in terms of our code - it's something going funky getting from the BRAW SDK to FxPlug4/FCPX.

**Comment by latenitefilms** [comment]
I've redesigned the Metal Pipeline to hopefully improve performance on your older Mac. If you're still having issues, please let me know in #25. Thanks!

**Comment by joema4** [comment]
I just tested 1.0.0 (26) on four-camera multicam using 6K BRAW on an M1 Ultra Mac Studio. Decode quality = full. Performance was good with all four angles up and Viewer set to Better Quality. It seems in the current version the BRAW parameters can only be altered during the import process, not after import. Is that correct?

**Reply by latenitefilms** [reply]
Great news! You can modify the BRAW Parameters after import by "going inside" the Synchronised Clip or Multicam Clip and going to the Effects tab.

See end of: https://github.com/latenitefilms/BRAWToolbox#installation
